Styling

namneeet opened the pull request:
Changed course page styling to make it more UI friendly
Added logout, add/deleting courses
Clicking on a course takes you to the particular course's page. Did the styling and added the components of viewing projects and adding students (Frontend for adding students needs to be integrated with the backend)
Fixed a few bugs and added an animated loading screen cuz why not

WMazerol committed:
Basic Metrics Experimentation

WMazerol committed:
Adds Commit Counter

WMazerol committed:
Metrics Frontend

WMazerol committed:
Fixing Merge Issues

WMazerol committed:
Create joy-rating-input.tsx

WMazerol committed:
Merge branch 'main' into feat-student-efficiency-metrics

WMazerol committed:
Revert "Merge branch 'main' into feat-student-efficiency-metrics"

This reverts commit 59c5647c658ec3f701165b5d8ce711f437dcdaa4, reversing
changes made to 96657039c6b67e2896641bf0846c87a0be66856d.

WMazerol committed:
Reorder Imports in Backend.py

WMazerol committed:
Fixes many merge issues

WMazerol committed:
Updated Metrics

WMazerol committed:
Merge branch 'main' into feat-student-efficiency-metrics

WMazerol committed:
Returns Joy Documents Instead of Dict

WMazerol committed:
Fixed Typo

WMazerol committed:
Update StudentMetrics.py

namneeet committed:
Merge branch 'Role-Access-Courses' into styling

namneeet committed:
fixed a bug

fixed some more bugs

JBBamba committed:
undid changes to dbwrapper + getTeamAVG method done

JBBamba committed:
Fixed my function

WMazerol committed:
Added Joy Graphs

WMazerol committed:
Merge branch 'feat-student-efficiency-metrics' of https://github.com/JackHuyn/SWE_4103_Gr3 into feat-student-efficiency-metrics

namneeet committed:
logout works

logout's done

WMazerol committed:
Set up Joy Ratings with DB Data

WMazerol committed:
Merge branch 'main' into feat-student-efficiency-metrics

namneeet committed:
layout for course page

layout for course page should be done, just need to connect the backend of add students to the front end, and need to view projects and students. Also need to be able to add projects

namneeet committed:
Merge branch 'main' into styling

namneeet committed:
Adding students for the front end and loading screen

adding students for the front end should be done and the backend was done too by raphael and sarun i believe so just integrating them together is left
also added a loading screen so it doesnt look ugly

namneeet committed:
fixed a few more bugs

fixed the error when u create a course and try to click on the newly added course

brennanleblanc0 reviewed (changes requested):
There are some conflicts with the current main branch

rferreir24 committed:
Merge branch 'styling' into Projects_page

Resolving conflicts with Styling

WMazerol committed:
Updated Metrics

WMazerol committed:
Merge branch 'main' into feat-student-efficiency-metrics

JBBamba committed:
Merge branch 'feat-student-efficiency-metrics' of https://github.com/JackHuyn/SWE_4103_Gr3 into feat-student-efficiency-metrics

WMazerol committed:
Update .gitignore

WMazerol committed:
Added Team Velocity Graph

WMazerol committed:
Added Team Velocity Endpoint

WMazerol committed:
Load Graphs onMount

WMazerol committed:
Github OAuth Requests

WMazerol committed:
Team Velocity Input

WMazerol committed:
Fixes Dialog Backdrop on Metrics Page

WMazerol committed:
Update

namneeet committed:
update

update

namneeet committed:
Merge remote-tracking branch 'origin/feat-student-efficiency-metrics' into styling

WMazerol committed:
Fixes Fb Authentication Issue

namneeet committed:
update

update

rferreir24 committed:
adding projects

Can Now:

1) Add a project (issues with refreshing)
2) Resolved 'styling' conflicts related to courseid

WMazerol committed:
Merge remote-tracking branch 'origin/Projects_page' into feat-student-efficiency-metrics

WMazerol committed:
Update courses.tsx

namneeet committed:
Merge remote-tracking branch 'origin/Projects_page' into styling

WMazerol committed:
Update courses.tsx

WMazerol committed:
User Class for User Sessions

WMazerol committed:
Merge branch 'main' into feat-complete-authentication

WMazerol committed:
Merge branch 'feat-complete-authentication' into feat-student-efficiency-metrics

WMazerol committed:
Update Backend.py

WMazerol committed:
Merge branch 'feat-complete-authentication' into feat-student-efficiency-metrics

rferreir24 committed:
refresh on courseid page works now

1) refresh works but slow as it first loads studentlist and then the projects (will be resolved later)

namneeet committed:
refresh fix

fixed the projects not displaying when u refresh the page

WMazerol committed:
GitHub Authentication

WMazerol committed:
Fixes Avg Joy Graph

WMazerol committed:
Fixes GitHub Metric Graphs

WMazerol committed:
Update Backend.py

rferreir24 committed:
added group page

1) added group page
2) moved and renamed file for dynamic routing

JBBamba committed:
Merge branch 'feat-student-efficiency-metrics' of https://github.com/JackHuyn/SWE_4103_Gr3 into feat-student-efficiency-metrics

rferreir24 committed:
Removed Buttons for student view

1) removed buttons for student views of projects and groups
2) Added projects and groups to middleware auth

rferreir24 committed:
Groups page

Removed form to create a new group

rferreir24 committed:
Added query in <Link> component

1) Added query in <Link> component so that we can retrieve the course_id for relevant fetches

namneeet committed:
Merge remote-tracking branch 'origin/Projects_page' into styling

namneeet committed:
css and stuff for the metrics

css for the metrics and group page and added a way to get the group id in the metrics page (only for the contributions graph though)

WMazerol committed:
Merge branch 'styling' into feat-student-efficiency-metrics

WMazerol committed:
Minor Fixes

JBBamba committed:
Merge branch 'feat-student-efficiency-metrics' of https://github.com/JackHuyn/SWE_4103_Gr3 into feat-student-efficiency-metrics

JBBamba committed:
dbwrapper joy operations

JBBamba committed:
Finished joy dbwrapper functions

namneeet committed:
update

update

namneeet committed:
Merge branch 'feat-student-efficiency-metrics' into styling

namneeet committed:
styling and new dialogue

added more styling and displays the github authorization dialogue when the status is 498

namneeet committed:
fixed some bugs

the authorize dialog wasnt displaying at the right times so thats fixed now, and just amde the styling a bit better, instead of a popup that covers the entire screen, its just a part of the github- related graphs
